src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -558,6 +558,14 @@ protected virtual BlockBlobClient GetNewInstanceBlockBlobClient(long version, st
             _logger.LogError(ex, message: "Access to storage account failed with ErrorCode: {ErrorCode}", ex.ErrorCode);
             throw new DataStoreRequestFailedException(ex, _blobClient.IsExternal);
         }
+        catch (AggregateException ex) when (_blobClient.IsExternal && ex.InnerException?.GetType() == typeof(RequestFailedException))
```

Comment:
In a aggregate exception, the inner Exception could be a list. Do you want to do something like

`aggEx.InnerExceptions.Any(x => x is RequestFailedException)`

Reply:
okay, atleast you could do ex.InnerException is RequestFailedException to simplify